### PR TITLE
Add support for pipeline

### PR DESCRIPTION
**Pull request by enwi**
This pull-request fixes the folowing issues
* [31899](https://issues.jenkins-ci.org/browse/JENKINS-31899)
* [41840](https://issues.jenkins-ci.org/browse/JENKINS-41840)

### CODE REVIEW

**Comment by rodrigc** [comment]
Good stuff!  I downloaded your branch and installed on my system which is Jenkins 2.46 with all the latest plugins.  I tried to use the Pipeline Snippet Generator to show the pipeline commands.

However the snippet generator only showed:

```
step <object of type org.korosoft.jenkins.plugin.rtp.RichTextPublisher>
```

**Comment by enwi** [comment]
I've got Jenkins 2.32.2 and everything works fine, but I know that when you use the snippet generator and select "step: General Build Step" and then select "Publish rich text message" the generator will show "step <object of type org.korosoft.jenkins.plugin.rtp.RichTextPublisher>", because I did not implement the functionality for "General Build Steps". On the other hand you can just select "rtp" as a sample step in the snippet generator and it should generate sth. like rtp parserName: 'WikiText', stableText: ''. Did you try that? Also maybe I should revise the code to make it not to show up at all in "General Build Steps".

**Comment by rodrigc** [comment]
Yes, you are absolutely correct.

- when I selected `rtp` as the sample step, I saw `rtp parserName: 'WikiText', stableText: ''`
- when I selected `step: General Build Step` and then `Publish rich text message`, then I saw `<object of type org.korosoft.jenkins.plugin.rtp.RichTextPublisher>`

I don't know what the convention is for implementing pipeline build steps, but
I think going in the direction to avoid confusion would be best.

**Comment by enwi** [comment]
Actually I just checked the code and also found this [tutorial](https://jenkins.io/blog/2016/05/25/update-plugin-for-pipeline/). So when you follow the tutorial the "General Build Step" should also work for rtp. I changed some code arround and will test if it will work now.

**Comment by enwi** [comment]
So I did a little bit of research and it seems that in "DescriptorImpl" of "RichTextPublisher" in function "FormValidation" 
`AbstractProject` 
needs to be repplaced by 
`Job` 
as well as 
`FilePath workspace = project.getSomeWorkspace();`
by
`FilePath workspace = new FilePath(project.getBuildDir());`
also the null check can be removed, but for me this does not work either. I stillr get
`<object of type org.korosoft.jenkins.plugin.rtp.RichTextPublisher>`

**Comment by enwi** [comment]
@rodrigc I fixed the snippet generator problem with [this commit].(f1eab8666de68ffbb9f1f0dd53f94b0f83309605)

File: src/main/java/org/korosoft/jenkins/plugin/rtp/RichTextPortlet.java [hunk 1]
```diff
@@ -71,26 +71,27 @@ public String getRichText() {
...
+            	return String.format(Messages.jobNotFound(), jobName);
             }
-            AbstractProject<?, ?> project = (AbstractProject<?, ?>) item;
+            
+            Job<?, ?> project = (Job<?, ?>) item;
             if (!useLastStable) {
                 return getRichTextFromActions(project.getActions(AbstractRichTextAction.class));
             } else {
-                for (AbstractBuild<?, ?> abstractBuild : project.getBuilds()) {
+                for (Run<?, ?> build : project.getBuilds()) {
```
Comment (masterhard) [inline src/main/java/org/korosoft/jenkins/plugin/rtp/RichTextPortlet.java]: run would be a better loop variable name here

File: src/main/java/org/korosoft/jenkins/plugin/rtp/RichTextPortlet.java [hunk 2]
```diff
@@ -71,26 +71,27 @@ public String getRichText() {
...
                 	
-                	Result res = abstractBuild.getResult();
+                	Result res = build.getResult();
                 	if(res == null) {
                 		throw new IllegalStateException("abstractBuild.getResult() is null!");
```
Comment (masterhard) [inline src/main/java/org/korosoft/jenkins/plugin/rtp/RichTextPortlet.java]: Exception message should be updated too

**Comment by enwi** [comment]
@masterhard I committed your suggested changes. Let me know if there is anything else that needs to be changed. I also set the regarding issues in jira to "in review".

**Comment by enwi** [comment]
@masterhard Just out of curiosity when will you merge this PR ?

**Comment by lechen26** [comment]
when this will be added on the pipeline?

**Comment by glib-briia** [comment]
Hi, Anyone got the idea when is it going to be released?

**Comment by masterhard** [comment]
I'm on it. Waiting for https://github.com/jenkins-infra/repository-permissions-updater/pull/291 approval now.

**Comment by masterhard** [comment]
Done

**Pull request merged**